tasks/bash.sh

```diff
@@ -25,18 +25,37 @@ success() {
   echo "$1"
 }
 
+determine_command_for_facter_4() {
+  puppet_version="$(puppet --version)"
+
+  if (( ${puppet_version%%.*} == 6 )); then
+    # puppet 6 with facter 4
+    facts_command="facter --json --show-legacy"
+  else
+    # puppet 7 with facter 4
+    facts_command="puppet facts show --show-legacy"
```

Comment:
I get an error running this that `show-legacy` isn't a valid option. I think we also want this to have `--render-as json` so that it returns json as expected.

Reply:
Did you try with puppet 7 or puppet 6?

With puppet 7:
```
puppet --version
7.0.0
puppet facts show --show-legacy
...
   "operatingsystemmajrelease":"20.04",
   "operatingsystemrelease":"20.04",
   "lsbdistrelease":"20.04",
   "lsbmajdistrelease":"20",
   "lsbminordistrelease":"04",
   "lsbdistcodename":"focal",
   "lsbdistid":"Ubuntu",
   "architecture":"amd64",
....
```

With puppet 6:
```
puppet --version
6.19.0
puppet facts show --show-legacy
Error: Could not parse application options: invalid option: --show-legacy
```

puppet 6 does not support `puppet facts show`.

Regarding `--render-as json` sure, the default it json at the moment, but we can add the flag for safety 👍